### PR TITLE
[GHA] Extend pre-commit by Mandatory API conformance on TEMPLATE

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *API mandatory scope on TEMPLATE should be 100%*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by mryzhov** [comment]
Why the Ubuntu pipeline has not been run?

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -419,6 +419,32 @@ jobs:
...
+            tar -czvf ${CONFORMANCE_ARTIFACTS_DIR}/conformance_artifacts.tar.gz *
+            popd
+
+        - name: Upload Conformance Artifacts
+          if: ${{ matrix.TEST_TYPE == "API" }}
+          uses: actions/upload-artifact@v3
+          with:
+            name: conformance_artifacts_${{ matrix.TEST_TYPE }}-TEMPLATE
+            path: ${{ env.CONFORMANCE_ARTIFACTS_DIR }}/conformance_artifacts.tar.gz
+            if-no-files-found: 'error'
```
Comment (akashchi) [inline .github/workflows/linux.yml:447]: It seems like the indentations are misaligned, make sure all the key names `- name`, `if`, `uses`, `run` are on the same level as they are in the previous steps.

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -419,6 +419,32 @@ jobs:
...
+
+      - name: Pack Conformance Artifacts
+        if: ${{ matrix.TEST_TYPE == "API" }}
+        run: |
+          pushd ${CONFORMANCE_ARTIFACTS_DIR}
+          tar -czvf ${CONFORMANCE_ARTIFACTS_DIR}/conformance_artifacts.tar.gz *
+          popd
+
+      - name: Upload Conformance Artifacts
+        if: ${{ matrix.TEST_TYPE == "API" }}
```
Comment (akashchi) [inline .github/workflows/linux.yml:442]:
```suggestion
        if: ${{ matrix.TEST_TYPE == 'API' }}
```

**Comment by iefode** [comment]
@akashchi Could you please approve the PR? CI is ready and contains new tests :)

**Review by akashchi — APPROVED** [review]

**Pull request merged**